### PR TITLE
Add README.md for coreHTTP library.

### PR DESCRIPTION
**Pull request by sarenameas**
Add README.md for the coreHTTP library. The language here matches the other FreeRTOS and AWS library repos.

This PR depends on https://github.com/FreeRTOS/coreHTTP/pull/28

Future:
- Additions about MISRA will update this README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,87 @@
-# HTTP Client Library
+# coreHTTP Client Library
 
 This repository contains a C language HTTP client library designed for embedded
```
Comment (aggarw13) [inline README.md:3]: Missing perdiod after the first sentence.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,87 @@
...
+platforms It has no dependencies on any additional libraries other than the
+standard C library and [http-parser](https://github.com/nodejs/http-parser), and
+a customer-implemented network transport interface. This library is distributed
+under the [MIT Open Source License](LICENSE).
+
+This library has gone through code quality checks including verification that no
+function has a [GNU Complexity](https://www.gnu.org/software/complexity/manual/complexity.html)
+score over 8. This library has also undergone both static code analysis from
+[Coverity static analysis](https://scan.coverity.com/), and validation of memory
+safety and proof of functional correctness through the
```
Comment (aggarw13) [inline README.md:13]: "proof of functional correctness" -> "data structure invariance" based on feedback from ARG team

File: README.md [hunk 3]
```diff
@@ -1,5 +1,87 @@
...
+[CBMC automated reasoning tool](https://www.cprover.org/cbmc/).
+
+## coreHTTP Config File
+
+The HTTP client library exposes build configuration macros that are required for
```
Comment (aggarw13) [inline README.md:18]: "The HTTP client library exposes ~~build~~ configuration macros that are required for building the library"

File: README.md [hunk 4]
```diff
@@ -1,5 +1,87 @@
...
+building the library. A list of all the configurations and their default values
+are defined in [core_http_config_defaults.h](source/include/core_http_config_defaults.h).
+To provide custom values for the configuration macros, a custom config file
+named `core_http_config.h` can be provided by the application to the library.
```
Comment (aggarw13) [inline README.md:22]: "..named `core_http_config.h` can be provided by the **user** application to the library."

File: README.md [hunk 5]
```diff
@@ -1,5 +1,87 @@
...
+* Defining the `HTTP_DO_NOT_USE_CUSTOM_CONFIG` preprocessor macro for the
+library build.
+
+## Building the Library
+
+The [httpFilePaths.cmake](httFilePaths.cmake) file contains the information of
+all source files and header include paths required to build the HTTP client
+library.
+
+As mentioned in the previous section, either a custom config file (i.e.
```
Comment (aggarw13) [inline README.md:42]:
```suggestion
As mentioned in the [previous section](#corehttp-config-file), either a custom config file (i.e.
```

File: README.md [hunk 6]
```diff
@@ -1,5 +1,87 @@
...
+[test/CMakeLists.txt](test/CMakeLists.txt) file.
+
+## Building Unit Tests
+
+### Platform Prerequisites
+
+- For running unit tests, the following are required:
+    - **C90 compiler** like gcc
+    - **CMake 3.13.0 or later**
+    - **Ruby 2.0.0 or later** is required for this repository's CMock test framework.
```
Comment (aggarw13) [inline README.md:57]: May be a hyperlink to the CMock repository would be good to have

File: README.md [hunk 7]
```diff
@@ -1,5 +1,87 @@
...
+
+1. Run this command to build the library and unit tests: `make -C build all`
+
+1. The generated test executables will be present in `build/bin/tests` folder.
+
+1. Run `cd build && ctest` to execute all tests and view the test run summary.
+
+## Reference examples
+
+The AWS IoT Embedded C-SDK repository contains demos of using the HTTP client
```
Comment (aggarw13) [inline README.md:76]:
```suggestion
The AWS IoT Device SDK for Embedded C repository contains demos of using the HTTP client
```

File: README.md [hunk 8]
```diff
@@ -1,5 +1,87 @@
...
 This repository contains a C language HTTP client library designed for embedded
-platforms. It is currently a work in progress and should not be used to create
-any products at the moment. We will update this README when that status changes.
\ No newline at end of file
+platforms It has no dependencies on any additional libraries other than the
+standard C library and [http-parser](https://github.com/nodejs/http-parser), and
+a customer-implemented network transport interface. This library is distributed
+under the [MIT Open Source License](LICENSE).
+
+This library has gone through code quality checks including verification that no
```
Comment (aggarw13) [inline README.md:9]: What about mentioning of MISRA deviations? Or are there some MISRA violations to be addressed?

File: README.md (hunk 2 above)
Comment (sarenameas) [reply README.md:13]: Ok this is copied from coreMQTT

File: README.md (hunk 1 above)
Comment (sarenameas) [reply README.md:3]: Thanks

File: README.md (hunk 5 above)
Comment (sarenameas) [reply README.md:42]: sure

File: README.md (hunk 8 above)
Comment (sarenameas) [reply README.md:9]: I will add that later. Forgot to put that in the description.

**Review by aggarw13 — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**